### PR TITLE
K8ssandra repository move

### PR DESCRIPTION
**Pull request by burmanm**
Add necessary changes for the new cass-operator repository to make it fit the k8ssandra structure:

* Module is now github.com/k8ssandra/cass-operator
* Modified paths in the go files
* Created new Github Actions that push k8ssandra/cass-operator as well as k8ssandra/system-logger
* New CODEOWNERS
* dockers builds use buildx with caching
* Go 1.16
* Automatic creation of images on every push to master

### CODE REVIEW

File: .github/workflows/operatorBuildAndDeploy.yml [hunk 1]
```diff
@@ -3,80 +3,76 @@ on:
   push:
     branches:
       - master
-      - "1.4.x"
```
Comment (jsanda) [inline .github/workflows/operatorBuildAndDeploy.yml:6]: Why did you remove the 1.4.x and 1.5.x branches?

File: operator/docker/base/Dockerfile [hunk 2]
```diff
@@ -67,53 +58,3 @@ ARG TARGETARCH
...
 
 CMD [ "/go/bin/operator" ]
-
-# UBI Image
-
-
-
-#############################################################
-
-FROM ${BASE_OS} AS builder-ubi
```
Comment (jsanda) [inline operator/docker/base/Dockerfile:77]: Are we building the UBI images elsewhere? If not then I am pretty sure we still need this. It is needed for OpenShift integration.

File: .github/workflows/operatorBuildAndDeploy.yml [hunk 3]
```diff
@@ -3,80 +3,131 @@ on:
...
-        env:
-          ECR_REPO: ${{ secrets.ECR_REPO }}
-          PR_REF: ${{ github.event.pull_request.head.ref }}
-          GITHUB_REPO_OWNER: ${{ github.repository_owner }}
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_PASSWORD }}
+      - name: Login to GitHub Package Registry
```
Comment (jsanda) [inline .github/workflows/operatorBuildAndDeploy.yml]: Are we still pushing images to the GH registry? Changes earlier in the file made me think we are not.

**Review by jsanda — CHANGES_REQUESTED** [review]

File: .github/workflows/operatorBuildAndDeploy.yml (hunk 1 above)
Comment (burmanm) [reply .github/workflows/operatorBuildAndDeploy.yml:6]: Removing old stuff we're not targeting.

Comment (jsanda) [reply .github/workflows/operatorBuildAndDeploy.yml:6]: I assume that those branches still need to be maintained for customer support. I suppose that they could be maintained in the repo in the DataStax org. Who is going to be handling bug fixes for customer support. I would expect over time that it will be us. If that is the case, does it make sense to keep them?

@bradfordcp can you chime in here?

File: operator/docker/base/Dockerfile (hunk 2 above)
Comment (burmanm) [reply operator/docker/base/Dockerfile:77]: What is our Openshift strategy? As far as I read from the cass-operator tickets, it hasn't been working for a while:

https://github.com/datastax/cass-operator/issues/266

So not in 4.5, not in 4.6, probably not in 4.7 .. and in k8ssandra we haven't tested anything against Openshift. Also, this k8ssandra/cass-operator path is not supported in the container catalog (only datastax/cass-operator is).

File: .github/workflows/operatorBuildAndDeploy.yml (hunk 3 above)
Comment (burmanm) [reply .github/workflows/operatorBuildAndDeploy.yml]: We can add it to as the last part of the push/buildx if we want. Don't know if we really need to anymore.

File: .github/workflows/operatorBuildAndDeploy.yml (hunk 1 above)
Comment (burmanm) [reply .github/workflows/operatorBuildAndDeploy.yml:6]: Wasn't that the deal that old versions are going to stay in the datastax/cass-operator?

File: .github/workflows/operatorBuildAndDeploy.yml (hunk 3 above)
Comment (jsanda) [reply .github/workflows/operatorBuildAndDeploy.yml]: If we don't need it, then can we remove this step?

File: operator/docker/base/Dockerfile (hunk 2 above)
Comment (jsanda) [reply operator/docker/base/Dockerfile:77]: @bradfordcp can you chime in here?

Comment (jsanda) [reply operator/docker/base/Dockerfile:77]: We can add back or update needed functionality in follow up PRs. I don't want to hold up merging on this as it currently does not impact k8ssandra.

File: .github/workflows/operatorBuildAndDeploy.yml (hunk 1 above)
Comment (jsanda) [reply .github/workflows/operatorBuildAndDeploy.yml:6]: We can add back or update needed functionality in follow up PRs. I don't want to hold up merging on this as it currently does not impact k8ssandra.

**Review by jsanda — APPROVED** [review]
